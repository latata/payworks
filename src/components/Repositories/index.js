import React from 'react';
import { Link } from 'react-router-dom';

function Repositories({ repositories }) {
  return (
    <table className="table" data-testid="repositories-list">
      <tbody>
      {repositories.map(repository => (
        <tr
          key={repository.name}>
          <td>
            <Link to={`/repository/${repository.full_name}`}>{repository.name}</Link>
          </td>
          <td>
            {repository.language}
          </td>
          <td>
            {repository.stargazers_count}
          </td>
        </tr>
      ))}
      </tbody>
    </table>
  );
}

export default Repositories;
