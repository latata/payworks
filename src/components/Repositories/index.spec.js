import React from 'react';
import { render } from 'react-testing-library';
import { MemoryRouter } from 'react-router-dom';

import Repositories from './index';

test('Repositories renders', () => {
  const repositories = [{
    name: 'repo1',
    full_name: 'Org/repo1',
    stargazers_count: 1,
    language: 'javascript',
  },{
    name: 'repo2',
    full_name: 'Org/repo2',
    stargazers_count: 2,
    language: 'javascript',
  }];
  const tree = render(<MemoryRouter><Repositories repositories={repositories} /></MemoryRouter>);
  expect(tree).toMatchSnapshot();
});
