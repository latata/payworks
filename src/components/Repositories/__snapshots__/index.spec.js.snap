// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table"
        data-testid="repositories-list"
      >
        <tbody>
          <tr>
            <td>
              <a
                href="/repository/Org/repo1"
              >
                repo1
              </a>
            </td>
            <td>
              javascript
            </td>
            <td>
              1
            </td>
          </tr>
          <tr>
            <td>
              <a
                href="/repository/Org/repo2"
              >
                repo2
              </a>
            </td>
            <td>
              javascript
            </td>
            <td>
              2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="table"
      data-testid="repositories-list"
    >
      <tbody>
        <tr>
          <td>
            <a
              href="/repository/Org/repo1"
            >
              repo1
            </a>
          </td>
          <td>
            javascript
          </td>
          <td>
            1
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/repository/Org/repo2"
            >
              repo2
            </a>
          </td>
          <td>
            javascript
          </td>
          <td>
            2
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
